server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java

```diff
@@ -92,6 +93,31 @@
      */
     volatile boolean primaryMode;
 
+    /**
+     * The current operation primary term. Management of this value is done through {@link IndexShard} and must only be done when safe. See
+     * {@link #setOperationPrimaryTerm(long)}.
+     */
+    private volatile long operationPrimaryTerm;
+
+    /**
+     * Returns the current operation primary term.
+     *
+     * @return the primary term
+     */
+    public long getOperationPrimaryTerm() {
```

Comment:
please move getter and setter further down. I understand this is your preferred coding style. However, it conflicts with the coding style we have adapted for all these classes and don't think this is the right time and place to switch things up.

Reply:
I pushed 72e2e7b15e7cc3d981f956c8b015e92ba7ccf19d.